src/formatter.hpp

```diff
@@ -67,6 +67,31 @@ class formatter
 	{
 		return stream_.str();
 	}
+	
+	// Support manipulators
+    formatter& operator<<(std::ostream&(*fn)(std::ostream&)) &
+	{
+        fn(stream_);
+		return *this;
+    }
+	
+    formatter&& operator<<(std::ostream&(*fn)(std::ostream&)) &&
+	{
+        fn(stream_);
+		return std::move(*this);
+    }
+	
+    formatter& operator<<(std::ios_base&(*fn)(std::ios_base&)) &
+	{
+        fn(stream_);
+		return *this;
+    }
+
+    formatter&& operator<<(std::ios_base&(*fn)(std::ios_base&)) &&
+	{
+        fn(stream_);
+		return std::move(*this);
+    }
```

Comment:
The indentation here is mixed up, please run it through clang-format.

Reply:
I think the force-push should fix this.